lighthouse-core/audits/seo/tap-targets.js

```diff
@@ -0,0 +1,274 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Checks that links, buttons, etc. are sufficiently large and don't overlap.
+ */
+const Audit = require('../audit');
+const ViewportAudit = require('../viewport');
+const {
+  getRectOverlapArea,
+  getRectAtCenter,
+  allRectsContainedWithinEachOther,
+  getLargestRect,
+} = require('../../lib/rect-helpers');
+const {getTappableRectsFromClientRects} = require('../../lib/tappable-rects');
+
+const FINGER_SIZE_PX = 48;
+// Ratio of the finger area tapping on an unintended element
+// to the finger area tapping on the intended element
+const MAX_ACCEPTABLE_OVERLAP_SCORE_RATIO = 0.25;
+
+
+/**
+ * @param {LH.Artifacts.Rect} cr
+ */
+function clientRectMeetsMinimumSize(cr) {
+  return cr.width >= FINGER_SIZE_PX && cr.height >= FINGER_SIZE_PX;
+}
+
+/**
+ * @param {LH.Artifacts.TapTarget} target
+ */
+function targetIsTooSmall(target) {
+  for (const cr of target.clientRects) {
+    if (clientRectMeetsMinimumSize(cr)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ *
+ * @param {LH.Artifacts.TapTarget[]} targets
+ */
+function getTooSmallTargets(targets) {
+  return targets.filter(targetIsTooSmall);
+}
+
+/**
+ *
+ * @param {LH.Artifacts.TapTarget[]} tooSmallTargets
+ * @param {LH.Artifacts.TapTarget[]} allTargets
+ */
+function getOverlapFailures(tooSmallTargets, allTargets) {
```

Comment:
nit: some naming changes would help make it clearer what calls what, throwing some ideas out there but I'm just looking for something that conveys the level of detail

is the below accurate-ish?
`getOverlapFailures` -> `getAllFailures`
`getTooCloseTargets` -> `getAllFailuresForTarget`
`getTargetTooCloseFailure` -> `getFailureForTargetPair`
`getOverlapFailure` -> `getFailureForClientRectPair`